etsin_finder/frontend/__tests__/qvain.apiv2.test.jsx

```diff
@@ -0,0 +1,1106 @@
+    mockQvain = createMockQvain({ original: { state: 'draft' } })
+    axios.post.mockReturnValue({
+      data: {
+        identifier: 'some identifier',
+      },
+    })
+    axios.patch.mockReturnValue({
+      data: {
+        identifier: 'some identifier',
+      },
+    })
+    Submit = new SubmitClass(mockQvain)
+  })
+
+  const expectNoError = async dataset => {
+    handleSubmitToBackend.mockReturnValue({ ...preparedDataset, ...dataset })
+    qvainFormSchema.validate.mockReturnValue(realQvainFormSchema.validate(dataset))
+
+    try {
+      await Submit.submitPublish()
+    } finally {
+      expect(Submit.error).toBe(undefined)
+    }
+  }
+
+  const expectError = async (dataset, error) => {
+    handleSubmitToBackend.mockReturnValue({ ...preparedDataset, ...dataset })
+    qvainFormSchema.validate.mockReturnValue(realQvainFormSchema.validate(dataset))
+    await Submit.submitPublish()
+    expect(Submit.error?.message).toEqual(error)
+  }
+
+  afterEach(async () => {
+    jest.resetAllMocks()
+  })
+
+  test('should call axios.patch and post', async () => {
+    handleSubmitToBackend.mockReturnValue(preparedDataset)
+    qvainFormSchema.validate.mockReturnValue(Promise.resolve(undefined))
+    await Submit.submitPublish()
+
+    expect(axios.patch).toHaveBeenCalledWith(
+      urls.v2.dataset(preparedDataset.original.identifier),
+      preparedDataset
+    )
+    expect(axios.post).toHaveBeenCalledWith(urls.v2.rpc.publishDataset(), null, {
+      params: { identifier: preparedDataset.original.identifier },
+    })
+  })
+
+  test('cases 55-57: no file origin, urn, cumulative state any', async () => {
+    const dataset = generateDefaultDatasetForPublish({ dataCatalog: undefined })
+
+    await expectError(dataset, errors.missingFileOrigin)
+  })
+
+  test('cases 58-60: no file origin, doi, cumulative state any', async () => {
+    const dataset = generateDefaultDatasetForPublish({ dataCatalog: undefined, useDoi: true })
+
+    await expectError(dataset, errors.missingFileOrigin)
+  })
+
+  test('case 61: ida, urn, cumulative state no', async () => {
+    const dataset = generateDefaultDatasetForPublish()
+
+    await expectNoError(dataset)
+  })
+
+  test('case 62: ida, urn, cumulative state yes', async () => {
+    const dataset = generateDefaultDatasetForPublish({ cumulativeState: CUMULATIVE_STATE.YES })
+
+    await expectNoError(dataset)
+  })
+
+  test('case 63: ida, urn, cumulative state yes', async () => {
+    const dataset = generateDefaultDatasetForPublish({ cumulativeState: CUMULATIVE_STATE.CLOSED })
+
+    await expectNoError(dataset)
+  })
+
+  test('case 64: ida, doi, cumulative state no', async () => {
+    const dataset = generateDefaultDatasetForPublish({ useDoi: true })
+
+    await expectNoError(dataset)
+  })
+
+  test('case 65: ida, doi, cumulative state yes', async () => {
+    const dataset = generateDefaultDatasetForPublish({
+      useDoi: true,
+      cumultaiveState: CUMULATIVE_STATE.YES,
```

Comment:
Typos in this and next ones
```suggestion
      cumulativeState: CUMULATIVE_STATE.YES,
```